include/mbedtls/aes.h

```diff
@@ -308,13 +308,22 @@ int mbedtls_aes_crypt_cfb8( mbedtls_aes_context *ctx,
  *        The OFB operation is identical for encryption or decryption, therefore
  *        no operation mode needs to be specified.
  *
- * \note  Upon exit, the content of the IV is updated so that you can
- *        call the same function again on the next
- *        block(s) of data and get the same result as if it was
- *        encrypted in one call. This allows a "streaming" usage.
- *        If you need to retain the contents of the
- *        IV, you must either save it manually or use the cipher
- *        module instead.
+ * \note  Upon exit, the content of iv, the Initialisation Vector, is updated
+ *        so that you can call the same function again on the next block(s) of
+ *        data and get the same result as if it was encrypted in one call. This
+ *        allows a "streaming" usage, by initialising iv_off to 0 before the
+ *        first call, and preserving its value between calls.
+ *
+ *        For block by block usage, (or non-streaming use), the iv should be
+ *        initialised on each call to a unique value, and iv_off set to 0 on
+ *        each call.
+ *
+ *        If you need to retain the contents of the initialisation vector, you
+ *        must either save it manually or use the cipher module instead.
+ *
+ *        For the OFB mode, the initiallisation vector must be unique and must
```

Comment:
Can we make this a warning? Also, I'm not sure I get "must be unique and must be unique for every encryption operation". Perhaps we should also advise that using a message counter is a fine way of achieving that.

Reply:
> Can we make this a warning?

Done.

> Also, I'm not sure I get "must be unique and must be unique for every encryption operation". Perhaps we should also advise that using a message counter is a fine way of achieving that.

As I said above, "it's ok for OFB to be predictable and use a message counter, but not for CBC or CFB. I think the documentation needs to be addressed there as well."

I don't think we can mention message counters in OFB without saying not to use them elsewhere. Mentioning it in OFB without also documenting the other block modes could lead to incorrect usage, where someone uses counter for CBC. I think it's better to raise an issue on the documentation to improve all the documentation for the block modes at once.